### PR TITLE
resource/aws_ec2_capacity_reservation: Fix to set ARN from its response and add owner_id attribute

### PR DESCRIPTION
**Pull request by shuheiktgw**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates https://github.com/hashicorp/terraform-provider-aws/issues/16978

Release note for [CHANGELOG](https://github.com/hashicorp/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
* resource/aws_ec2_capacity_reservation: Add owner_id attribute
```

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccAWSEc2CapacityReservation_*'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSEc2CapacityReservation_* -timeout 120m
=== RUN   TestAccAWSEc2CapacityReservation_basic
=== PAUSE TestAccAWSEc2CapacityReservation_basic
=== RUN   TestAccAWSEc2CapacityReservation_ebsOptimized
=== PAUSE TestAccAWSEc2CapacityReservation_ebsOptimized
=== RUN   TestAccAWSEc2CapacityReservation_endDate
=== PAUSE TestAccAWSEc2CapacityReservation_endDate
=== RUN   TestAccAWSEc2CapacityReservation_endDateType
=== PAUSE TestAccAWSEc2CapacityReservation_endDateType
=== RUN   TestAccAWSEc2CapacityReservation_ephemeralStorage
=== PAUSE TestAccAWSEc2CapacityReservation_ephemeralStorage
=== RUN   TestAccAWSEc2CapacityReservation_instanceCount
=== PAUSE TestAccAWSEc2CapacityReservation_instanceCount
=== RUN   TestAccAWSEc2CapacityReservation_instanceMatchCriteria
=== PAUSE TestAccAWSEc2CapacityReservation_instanceMatchCriteria
=== RUN   TestAccAWSEc2CapacityReservation_instanceType
=== PAUSE TestAccAWSEc2CapacityReservation_instanceType
=== RUN   TestAccAWSEc2CapacityReservation_tags
=== PAUSE TestAccAWSEc2CapacityReservation_tags
=== RUN   TestAccAWSEc2CapacityReservation_disappears
=== PAUSE TestAccAWSEc2CapacityReservation_disappears
=== RUN   TestAccAWSEc2CapacityReservation_tenancy
    provider_test.go:56: EC2 Capacity Reservations do not currently support dedicated tenancy.
--- SKIP: TestAccAWSEc2CapacityReservation_tenancy (0.00s)
=== CONT  TestAccAWSEc2CapacityReservation_basic
=== CONT  TestAccAWSEc2CapacityReservation_instanceMatchCriteria
=== CONT  TestAccAWSEc2CapacityReservation_disappears
=== CONT  TestAccAWSEc2CapacityReservation_instanceType
=== CONT  TestAccAWSEc2CapacityReservation_endDateType
=== CONT  TestAccAWSEc2CapacityReservation_instanceCount
=== CONT  TestAccAWSEc2CapacityReservation_tags
=== CONT  TestAccAWSEc2CapacityReservation_ephemeralStorage
=== CONT  TestAccAWSEc2CapacityReservation_endDate
=== CONT  TestAccAWSEc2CapacityReservation_ebsOptimized
--- PASS: TestAccAWSEc2CapacityReservation_disappears (50.26s)
--- PASS: TestAccAWSEc2CapacityReservation_ebsOptimized (54.86s)
--- PASS: TestAccAWSEc2CapacityReservation_instanceMatchCriteria (59.14s)
--- PASS: TestAccAWSEc2CapacityReservation_ephemeralStorage (59.20s)
--- PASS: TestAccAWSEc2CapacityReservation_basic (59.57s)
--- PASS: TestAccAWSEc2CapacityReservation_instanceType (81.62s)
--- PASS: TestAccAWSEc2CapacityReservation_instanceCount (84.38s)
--- PASS: TestAccAWSEc2CapacityReservation_endDate (84.71s)
--- PASS: TestAccAWSEc2CapacityReservation_tags (104.90s)
--- PASS: TestAccAWSEc2CapacityReservation_endDateType (107.50s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	109.433s
```

Thank you for your review! 👍

### CODE REVIEW

File: aws/resource_aws_ec2_capacity_reservation_test.go [hunk 1]
```diff
@@ -88,6 +88,7 @@ func TestAccAWSEc2CapacityReservation_basic(t *testing.T) {
 					resource.TestCheckResourceAttr(resourceName, "instance_match_criteria", "open"),
 					resource.TestCheckResourceAttr(resourceName, "instance_platform", "Linux/UNIX"),
 					resource.TestCheckResourceAttr(resourceName, "instance_type", "t2.micro"),
+					resource.TestCheckResourceAttrSet(resourceName, "owner_id"),
```
Comment (ewbankkit) [inline aws/resource_aws_ec2_capacity_reservation_test.go:91]: Can verify that the `owner_id` is set to the current AWS account ID.

```suggestion
					testAccCheckResourceAttrAccountID(resourceName, "owner_id"),
```

Comment (shuheiktgw) [reply aws/resource_aws_ec2_capacity_reservation_test.go:91]: Didn't know that kind of test helper exists. Thank you!

**Comment by shuheiktgw** [comment]
```
make testacc TESTARGS='-run=TestAccAWSEc2CapacityReservation_*'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSEc2CapacityReservation_* -timeout 120m
=== RUN   TestAccAWSEc2CapacityReservation_basic
=== PAUSE TestAccAWSEc2CapacityReservation_basic
=== RUN   TestAccAWSEc2CapacityReservation_ebsOptimized
=== PAUSE TestAccAWSEc2CapacityReservation_ebsOptimized
=== RUN   TestAccAWSEc2CapacityReservation_endDate
=== PAUSE TestAccAWSEc2CapacityReservation_endDate
=== RUN   TestAccAWSEc2CapacityReservation_endDateType
=== PAUSE TestAccAWSEc2CapacityReservation_endDateType
=== RUN   TestAccAWSEc2CapacityReservation_ephemeralStorage
=== PAUSE TestAccAWSEc2CapacityReservation_ephemeralStorage
=== RUN   TestAccAWSEc2CapacityReservation_instanceCount
=== PAUSE TestAccAWSEc2CapacityReservation_instanceCount
=== RUN   TestAccAWSEc2CapacityReservation_instanceMatchCriteria
=== PAUSE TestAccAWSEc2CapacityReservation_instanceMatchCriteria
=== RUN   TestAccAWSEc2CapacityReservation_instanceType
=== PAUSE TestAccAWSEc2CapacityReservation_instanceType
=== RUN   TestAccAWSEc2CapacityReservation_tags
=== PAUSE TestAccAWSEc2CapacityReservation_tags
=== RUN   TestAccAWSEc2CapacityReservation_disappears
=== PAUSE TestAccAWSEc2CapacityReservation_disappears
=== RUN   TestAccAWSEc2CapacityReservation_tenancy
    provider_test.go:56: EC2 Capacity Reservations do not currently support dedicated tenancy.
--- SKIP: TestAccAWSEc2CapacityReservation_tenancy (0.00s)
=== CONT  TestAccAWSEc2CapacityReservation_basic
=== CONT  TestAccAWSEc2CapacityReservation_instanceMatchCriteria
=== CONT  TestAccAWSEc2CapacityReservation_tags
=== CONT  TestAccAWSEc2CapacityReservation_instanceType
=== CONT  TestAccAWSEc2CapacityReservation_endDateType
=== CONT  TestAccAWSEc2CapacityReservation_instanceCount
=== CONT  TestAccAWSEc2CapacityReservation_ephemeralStorage
=== CONT  TestAccAWSEc2CapacityReservation_endDate
=== CONT  TestAccAWSEc2CapacityReservation_disappears
=== CONT  TestAccAWSEc2CapacityReservation_ebsOptimized
--- PASS: TestAccAWSEc2CapacityReservation_disappears (55.56s)
--- PASS: TestAccAWSEc2CapacityReservation_instanceMatchCriteria (63.25s)
--- PASS: TestAccAWSEc2CapacityReservation_ephemeralStorage (64.15s)
--- PASS: TestAccAWSEc2CapacityReservation_ebsOptimized (65.51s)
--- PASS: TestAccAWSEc2CapacityReservation_basic (65.75s)
--- PASS: TestAccAWSEc2CapacityReservation_instanceCount (94.27s)
--- PASS: TestAccAWSEc2CapacityReservation_endDate (96.28s)
--- PASS: TestAccAWSEc2CapacityReservation_instanceType (106.68s)
--- PASS: TestAccAWSEc2CapacityReservation_tags (118.39s)
--- PASS: TestAccAWSEc2CapacityReservation_endDateType (120.74s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	122.683s
```

**Comment by shuheiktgw** [comment]
@ewbankkit Fixed it so would you review the PR again? 🙏

**Review by ewbankkit — APPROVED** [review]
LGTM.

```console
$ make testacc TEST=./aws TESTARGS='-run=TestAccAWSEc2CapacityReservation_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSEc2CapacityReservation_ -timeout 120m
=== RUN   TestAccAWSEc2CapacityReservation_basic
=== PAUSE TestAccAWSEc2CapacityReservation_basic
=== RUN   TestAccAWSEc2CapacityReservation_ebsOptimized
=== PAUSE TestAccAWSEc2CapacityReservation_ebsOptimized
=== RUN   TestAccAWSEc2CapacityReservation_endDate
=== PAUSE TestAccAWSEc2CapacityReservation_endDate
=== RUN   TestAccAWSEc2CapacityReservation_endDateType
=== PAUSE TestAccAWSEc2CapacityReservation_endDateType
=== RUN   TestAccAWSEc2CapacityReservation_ephemeralStorage
=== PAUSE TestAccAWSEc2CapacityReservation_ephemeralStorage
=== RUN   TestAccAWSEc2CapacityReservation_instanceCount
=== PAUSE TestAccAWSEc2CapacityReservation_instanceCount
=== RUN   TestAccAWSEc2CapacityReservation_instanceMatchCriteria
=== PAUSE TestAccAWSEc2CapacityReservation_instanceMatchCriteria
=== RUN   TestAccAWSEc2CapacityReservation_instanceType
=== PAUSE TestAccAWSEc2CapacityReservation_instanceType
=== RUN   TestAccAWSEc2CapacityReservation_tags
=== PAUSE TestAccAWSEc2CapacityReservation_tags
=== RUN   TestAccAWSEc2CapacityReservation_disappears
=== PAUSE TestAccAWSEc2CapacityReservation_disappears
=== RUN   TestAccAWSEc2CapacityReservation_tenancy
    provider_test.go:56: EC2 Capacity Reservations do not currently support dedicated tenancy.
--- SKIP: TestAccAWSEc2CapacityReservation_tenancy (0.00s)
=== CONT  TestAccAWSEc2CapacityReservation_basic
=== CONT  TestAccAWSEc2CapacityReservation_disappears
=== CONT  TestAccAWSEc2CapacityReservation_tags
=== CONT  TestAccAWSEc2CapacityReservation_instanceType
=== CONT  TestAccAWSEc2CapacityReservation_instanceMatchCriteria
=== CONT  TestAccAWSEc2CapacityReservation_instanceCount
=== CONT  TestAccAWSEc2CapacityReservation_ephemeralStorage
=== CONT  TestAccAWSEc2CapacityReservation_endDateType
=== CONT  TestAccAWSEc2CapacityReservation_endDate
=== CONT  TestAccAWSEc2CapacityReservation_ebsOptimized
--- PASS: TestAccAWSEc2CapacityReservation_disappears (32.18s)
--- PASS: TestAccAWSEc2CapacityReservation_ephemeralStorage (37.22s)
--- PASS: TestAccAWSEc2CapacityReservation_ebsOptimized (37.94s)
--- PASS: TestAccAWSEc2CapacityReservation_basic (38.23s)
--- PASS: TestAccAWSEc2CapacityReservation_instanceMatchCriteria (38.24s)
--- PASS: TestAccAWSEc2CapacityReservation_endDate (51.76s)
--- PASS: TestAccAWSEc2CapacityReservation_instanceType (51.93s)
--- PASS: TestAccAWSEc2CapacityReservation_instanceCount (52.87s)
--- PASS: TestAccAWSEc2CapacityReservation_endDateType (64.23s)
--- PASS: TestAccAWSEc2CapacityReservation_tags (64.51s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	64.607s
```

File: aws/resource_aws_ec2_capacity_reservation.go [hunk 2]
```diff
@@ -205,22 +208,14 @@ func resourceAwsEc2CapacityReservationRead(d *schema.ResourceData, meta interfac
...
 
 	d.Set("tenancy", reservation.Tenancy)
-
-	arn := arn.ARN{
-		Partition: meta.(*AWSClient).partition,
-		Service:   "ec2",
-		Region:    meta.(*AWSClient).region,
-		AccountID: meta.(*AWSClient).accountid,
-		Resource:  fmt.Sprintf("capacity-reservation/%s", d.Id()),
-	}.String()
```
Comment (bflad) [inline aws/resource_aws_ec2_capacity_reservation.go:221]: When removing this type of manual ARN building, we should double check that it is not in the `website/docs/index.html` file `skip_requesting_account_id` list of resources that are documented as potentially not working. 👍

**Review by bflad — APPROVED** [review]
Super happy to see these in the API. Thank you, @shuheiktgw 🚀 

Output from acceptance testing in AWS Commercial:

```
--- PASS: TestAccAWSEc2CapacityReservation_disappears (21.93s)
--- PASS: TestAccAWSEc2CapacityReservation_ebsOptimized (25.87s)
--- PASS: TestAccAWSEc2CapacityReservation_instanceMatchCriteria (25.99s)
--- PASS: TestAccAWSEc2CapacityReservation_ephemeralStorage (26.00s)
--- PASS: TestAccAWSEc2CapacityReservation_basic (26.39s)
--- PASS: TestAccAWSEc2CapacityReservation_endDate (37.97s)
--- PASS: TestAccAWSEc2CapacityReservation_instanceCount (38.20s)
--- PASS: TestAccAWSEc2CapacityReservation_instanceType (38.55s)
--- PASS: TestAccAWSEc2CapacityReservation_endDateType (48.65s)
--- PASS: TestAccAWSEc2CapacityReservation_tags (49.90s)
```

Output from acceptance testing in AWS GovCloud (US):

```
--- PASS: TestAccAWSEc2CapacityReservation_disappears (25.87s)
--- PASS: TestAccAWSEc2CapacityReservation_instanceMatchCriteria (28.03s)
--- PASS: TestAccAWSEc2CapacityReservation_ephemeralStorage (28.90s)
--- PASS: TestAccAWSEc2CapacityReservation_ebsOptimized (29.25s)
--- PASS: TestAccAWSEc2CapacityReservation_basic (29.76s)
--- PASS: TestAccAWSEc2CapacityReservation_instanceType (42.88s)
--- PASS: TestAccAWSEc2CapacityReservation_instanceCount (44.02s)
--- PASS: TestAccAWSEc2CapacityReservation_endDate (44.05s)
--- PASS: TestAccAWSEc2CapacityReservation_endDateType (57.51s)
--- PASS: TestAccAWSEc2CapacityReservation_tags (58.11s)
```

**Pull request merged**